### DISCUSSION TITLE
Needing some specific functionality

### DISCUSSION THREAD

**Discussion by JulieVelghe** [verbb/formie#1692] [Q&A]
Hi all, 
I have website for a client, that is a craft multisite setup. I need to setup some forms on one of them, but I run into some functional issues. It's a school website, with the subsite being a sort of students platform. A contactform on that subsite needs to access some data on the main site.

- entries-fields seem to only take entries from the current site. Is there a way to be able to select entries from any other sites in the same craft setup? For example, the main site has the list of courses, and the form on the subsites needs to be able to ge those.
- entries-fields are not limited to the current user. Is there some way to do this?
- linked files in a file upload field cannot be removed, only swapped. Is there a way to be able to remove a file from a file upload field? 

Perhaps some of these things are possible via a custom module? Any help is appreciated.

**Comment by engram-design** [comment]
As forms are multi-site specific at the moment, you'll have to achieve some of these things through a custom module, or custom templates where you modify the entries element query.

There's a bit of a [user guide](https://verbb.io/craft-plugins/formie/user-guides/how-to-modify-element-fields-queries-to-change-what-elements-can-be-chosen) on this. Here, you can change the query to fetch entries for the site you want.

```twig
{% do field.setElementsQuery(elementsQuery.siteId(2)) %}
```

As for not being limited to the current user, that's not something Formie deals with, as there's not always a user to restrict by - to handle guests for instance.

As for file upload fields, that's indeed not possible at the moment, as is more a browser restriction than anything else. We have plans to include a more [feature-rich](https://github.com/verbb/formie/issues/133) component.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.6](https://github.com/verbb/formie/releases/tag/4.0.0-beta.6).
